NewEssentials/Players/AfkChecker.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Cysharp.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using NewEssentials.API.Players;
+using NewEssentials.Extensions;
+using OpenMod.API.Eventing;
+using OpenMod.API.Ioc;
+using OpenMod.API.Plugins;
+using OpenMod.API.Users;
+using OpenMod.Core.Helpers;
+using OpenMod.Core.Users;
+using OpenMod.Core.Users.Events;
+using OpenMod.Unturned.Users;
+using SDG.Unturned;
+
+namespace NewEssentials.Players
+{
+    [ServiceImplementation]
+    public class AfkChecker : IAfkChecker, IAsyncDisposable
+    {
+        
+        //TODO: Add translations
+
+        private Dictionary<IUser, TimeSpan> m_Users;
+
+        private readonly IEventBus m_Events;
+        private readonly IConfiguration m_Config;
+        private readonly NewEssentials m_Plugin;
+        private readonly IUserManager m_UserManager;
+        
+        public AfkChecker(IEventBus bus, IConfiguration config, IPluginAccessor<NewEssentials> plugin, IUserManager users)
+        {
+            if (!config.GetValue<bool>("afkchecker:enabled"))
+                return;
+            m_Users = new Dictionary<IUser, TimeSpan>();
+            
+            bus.Subscribe<UserConnectedEvent>(plugin.Instance, (provider, sender, @event) => PlayerJoin(@event));
```

Comment:
Subscribe to events by injecting and using IRuntime instead of plugin.Instance to stay as long as subscribed as OpenMod is loaded. Otherwise reloads will not be supported because the plugin will receive a new instance this instance will have set IsComponentAlive set to false.